File: Java/Strategy/src/main/java/MallardDuck.java
package duck.strategy;

public class MallardDuck extends Duck {
	//Constructor
	public MallardDuck() {
		quackBehavior =  new Quack();
		flyBehaviour = new FlyWithWings();
	}

	public void display() {
		System.out.println("I'm a real Mallard Duck");
	}

}

File: Cpp/hfdp/src/strategy/ducks/MallardDuck.cpp
//
// Created by Johnson, Chase on 12/4/16.
//

#include <iostream>
#include "MallardDuck.h"
#include "Quack.h"
#include "FlyWithWings.h"

MallardDuck::MallardDuck() {
    this->setFlyBehaviour(new FlyWithWings());
    this->setQuackBehaviour(new Quack());
}

void MallardDuck::display() {
    std::cout << "I'm a real Mallard Duck" << std::endl;
}
